azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java

```diff
@@ -219,6 +219,27 @@ public ExecutableFlow fetchExecutableFlow(int id)
   }
 
   @Override
+  public Pair<ExecutionReference, ExecutableFlow> fetchActiveFlowByExecId(int execId)
+      throws ExecutorManagerException {
+    QueryRunner runner = createQueryRunner();
+    FetchActiveExecutableFlowByExecId flowHandler = new FetchActiveExecutableFlowByExecId();
+
+    try {
+      List<Pair<ExecutionReference, ExecutableFlow>> flows =
+          runner.query(FetchActiveExecutableFlowByExecId.FETCH_ACTIVE_EXECUTABLE_FLOW_BY_EXECID,
+              flowHandler, execId);
+      if(flows.isEmpty()) {
+        return null;
+      }
+      else {
+        return flows.get(0);
+      }
+    } catch (SQLException e) {
+      throw new ExecutorManagerException("Error fetching active flows by exec id", e);
+    }
+  }
+
+  @Override
   public int fetchNumExecutableFlows() throws ExecutorManagerException {
```

Comment:
You are right. fetchJobInfoAttempts is never used. Also I double checked fetchNumExecutableFlows. Actually fetchNumExecutableFlows(int projectId, String flowId) is used but fetchNumExecutableFlows() is not. I'll remove the unused ones.